js/packages/teams-ai/src/Application.ts

```diff
@@ -747,6 +747,111 @@ export class Application<TState extends TurnState = DefaultTurnState> {
     }
 }
 
+/**
+ * A builder class for simplifying the creation of an Application instance.
+ * @template TState Optional. Type of the turn state. This allows for strongly typed access to the turn state.
+ */
+export class ApplicationBuilder<TState extends TurnState = DefaultTurnState> {
+    private _options: Partial<ApplicationOptions<TState>> = {};
+
+    /**
+     * Configures the application to use long running messages.
+     * @param {BotAdapter} adapter The adapter to use for routing incoming requests.
+     * @param {string} botAppId The Microsoft App ID for the bot.
+     * @returns {this} The ApplicationBuilder instance.
+     */
+    public withLongRunningMessages(adapter: BotAdapter, botAppId: string): this {
+        if (!botAppId) {
+            throw new Error(
+                `The Application.longRunningMessages property is unavailable because no botAppId was configured.`
+            );
+        }
+
+        this._options.longRunningMessages = true;
+        this._options.adapter = adapter;
+        this._options.botAppId = botAppId;
+        return this;
+    }
+
+    /**
+     * Configures the storage system to use for storing the bot's state.
+     * @param {Storage} storage The storage system to use.
+     * @returns {this} The ApplicationBuilder instance.
+     */
+    public withStorage(storage: Storage): this {
+        this._options.storage = storage;
+        return this;
+    }
+
+    /**
+     * Configures the AI system to use for processing incoming messages.
+     * @param {AIOptions<TState>} aiOptions The options for the AI system.
+     * @returns {this} The ApplicationBuilder instance.
+     */
+    public withAI(aiOptions: AIOptions<TState>): this {
+        this._options.ai = aiOptions;
+        return this;
+    }
+
+    /**
+     * Configures the turn state manager to use for managing the bot's turn state.
+     * @param {TurnStateManager<TState>} turnStateManager The turn state manager to use.
+     * @returns {this} The ApplicationBuilder instance.
+     */
+    public withTurnStateManager(turnStateManager: TurnStateManager<TState>): this {
+        this._options.turnStateManager = turnStateManager;
+        return this;
+    }
+
+    /**
+     * Configures the processing of Adaptive Card requests.
+     * @param {AdaptiveCardsOptions} adaptiveCardOptions The options for the Adaptive Cards.
+     * @returns {this} The ApplicationBuilder instance.
+     */
+    public withAdaptiveCards(adaptiveCardOptions: AdaptiveCardsOptions): this {
```

Comment:
What about Message Extensions?

Reply:
Nice check, message extensions doesn't currently have a `messageExtensionsOptions` and is also not configurable in the `Application `class